services/ui_backend_service/api/dag.py

```diff
@@ -43,24 +43,11 @@ async def get_run_dag(self, request):
                 schema:
                     $ref: '#/definitions/ResponsesDagError500'
         """
-        flow_name = request.match_info['flow_id']
-        run_id_key, run_id_value = translate_run_key(
-            request.match_info.get("run_number"))
-        # 'code-package' value contains json with dstype, sha1 hash and location
-        # 'code-package-url' value contains only location as a string
-        db_response, *_ = await self._metadata_table.find_records(
-            conditions=[
-                "flow_id = %s",
-                "{run_id_key} = %s".format(
-                    run_id_key=run_id_key),
-                "(field_name = %s OR field_name = %s)"
-            ],
-            values=[
-                flow_name, run_id_value,
-                "code-package", "code-package-url"
-            ],
-            fetch_single=True, expanded=True
-        )
+        db_response = await self.get_run_ids_from_graph_info_artifact(request)
+        if not db_response.response_code == 200:
+            # Try to look for codepackage if graph artifact is missing
+            db_response = await self.get_run_ids_from_codepackage_metadata(request)
+
```

Comment:
Hmm I don't entirely follow the logic here, can you add a comment explaining it? 

I _think_ the original version of this code would essentially simplify verify that the run was executed on a remote backend and has a code package (otherwise getting constructing the graph would be impossible). Now we want to check that either the run has `_graph_info` or has the code package.

If that's true, worth adding a comment explaining all that and maybe naming the functions something like `check_run_has_graph_info` or similar, since its not literally getting run_id from the artifact here

Reply:
cleaned up the structure a bit and added comments. You're correct in the general logic, we want to check before firing the cache action, whether the required data for a DAG actually exists. Now there is a clear fallback where we check for `_graph_info` artifact -> `code-package` metadata -> 404 if neither exist.